### PR TITLE
type definitions for imagemin 7.0

### PR DESCRIPTION
**Pull request by hkjeffchan**
update to be compatible for old version of other packages

### CODE REVIEW

**Comment by romain-faust** [comment]
@hkjeffchan I see you've opened another PR https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38193 should I review both or only this one ?

**Comment by hkjeffchan** [comment]
> @hkjeffchan I see you've opened another PR #38193 should I review both or only this one ?

Please ignore the old one. Thanks.

File: types/imagemin/index.d.ts [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
 // Project: https://github.com/imagemin/imagemin#readme
 // Definitions by: Romain Faust <https://github.com/romain-faust>
+//                 Jeff Chan <https://github.com/hkjeffchan>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
 
-declare function imagemin(input: ReadonlyArray<string>, outputOrOptions?: string | imagemin.Options): Promise<imagemin.Result[]>;
-declare function imagemin(input: ReadonlyArray<string>, output?: string, options?: imagemin.Options): Promise<imagemin.Result[]>;
+declare function imagemin(input: string[], options?: imagemin.Options): Promise<imagemin.Result[]>;
```
Comment (romain-faust) [inline types/imagemin/index.d.ts]: Any reason you've replaced `ReadonlyArray<string>` with `string[]` ?

File: types/imagemin/index.d.ts [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
 
 declare namespace imagemin {
     type Plugin = (input: Buffer) => Promise<Buffer>;
 
     interface Options {
```
Comment (romain-faust) [inline types/imagemin/index.d.ts]: You should create a separate type for `buffer`'s options. In [the documentation](https://github.com/imagemin/imagemin#options-1) the buffer options does not accept the `destination` and `glob` properties

**Review by romain-faust — CHANGES_REQUESTED** [review]

File: types/imagemin/index.d.ts (hunk 1 above)
Comment (hkjeffchan) [reply types/imagemin/index.d.ts]: No, you may keep it ReadonlyArray<string>

File: types/imagemin/index.d.ts (hunk 2 above)
Comment (hkjeffchan) [reply types/imagemin/index.d.ts]: Sorry, I did so in PR #38193 and it seems it fails some testing.

Comment (hkjeffchan) [reply types/imagemin/index.d.ts]: Updated with BufferOption

**Comment by hkjeffchan** [comment]
@romain-faust Please check with this one. Thanks.

**Review by romain-faust — APPROVED** [review]
LGTM

File: types/imagemin/index.d.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ declare namespace imagemin {
...
-        destination?: string;
+        destination: string;
```
Comment (romain-faust) [inline types/imagemin/index.d.ts]: [`destination` should be optional](https://github.com/imagemin/imagemin#destination)

**Review by romain-faust — CHANGES_REQUESTED** [review]

**Review by hkjeffchan — COMMENTED** [review]
Yes, you are correct. Just tested and it is optional. Sorry for the chaos.

File: types/imagemin/index.d.ts (hunk 3 above)
Comment (hkjeffchan) [reply types/imagemin/index.d.ts]: Yes, you are correct. Just tested and it is optional. Sorry for the chaos.

Comment (hkjeffchan) [reply types/imagemin/index.d.ts]: Sorry, you need to approve it again. I reverted last commit. @romain-faust

**Comment by romain-faust** [comment]
@hkjeffchan don't worry ;)

**Review by romain-faust — APPROVED** [review]

**Pull request merged**